wolfcrypt/src/asn.c

```diff
@@ -5508,7 +5508,7 @@ int wc_RsaPrivateKeyDecode(const byte* input, word32* inOutIdx, RsaKey* key,
     DECL_ASNGETDATA(dataASN, rsaKeyASN_Length);
     int        ret = 0;
     byte       i;
-    byte       version;
+    byte       version = (byte)-1;
```

Comment:
Is the cast necessary?

Reply:
Yes cast is needed. I would prefer `0xff` thought. Also @douzzer can you squash the 34 commits? Or at least squash the CPPcheck vs LinuxKM move into two commits.

Reply:
I left them unsquashed to facilitate bisecting in case one of the zillions of point changes caused a problem not detected by testing.